### PR TITLE
[Lang] Support struct fors on ti.any_arr

### PR DESCRIPTION
**Pull request by strongoier**
Related issue = #2772

### CODE REVIEW

**Review by ailzhang — APPROVED** [review]
Looks great! Thanks!

File: taichi/ir/frontend_ir.h [hunk 1]
```diff
@@ -392,9 +392,13 @@ class ExternalTensorExpression : public Expression {
   DataType dt;
   int dim;
   int arg_id;
+  int element_dim;
```
Comment (k-ye) [inline taichi/ir/frontend_ir.h:395]: nit1: Add comment saying that this encodes SOA/SOA
nit2: init these members

**Review by k-ye — APPROVED** [review]
LGTM!

File: taichi/ir/frontend_ir.h (hunk 1 above)
Comment (strongoier) [reply taichi/ir/frontend_ir.h]: nit1: done
nit2: I can't think of any meaningful initial values for these members.. Other expressions don't init them either.

**Pull request merged**